examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjectsStorage.agent.ts

```diff
@@ -22,44 +22,30 @@ export class BpiSubjectStorageAgent extends PrismaService {
     if (!bpiSubjectModel) {
       throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
     }
-    return this.mapper.map(
-      bpiSubjectModel,
-      getType<BpiSubject>(),
-    ) as BpiSubject;
+    return this.mapper.map(bpiSubjectModel, BpiSubject, BpiSubject);
   }
 
   async getAllBpiSubjects(): Promise<BpiSubject[]> {
     const bpiSubjectModels = await this.bpiSubject.findMany();
     return bpiSubjectModels.map((bpiSubjectModel) => {
-      return this.mapper.map(
-        bpiSubjectModel,
-        getType<BpiSubject>(),
-      ) as BpiSubject;
+      return this.mapper.map(bpiSubjectModel, BpiSubject, BpiSubject);
     });
   }
 
   async createNewBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
     const newBpiSubjectModel = await this.bpiSubject.create({
-      data: this.mapper.map(bpiSubject, getType<BpiSubject>()) as BpiSubject,
+      data: bpiSubject,
     });
 
-    return this.mapper.map(
-      newBpiSubjectModel,
-      getType<BpiSubject>(),
-    ) as BpiSubject;
+    return this.mapper.map(newBpiSubjectModel, BpiSubject, BpiSubject);
   }
 
   async updateBpiSubject(bpiSubject: BpiSubject): Promise<BpiSubject> {
     const updatedBpiSubjectModel = await this.bpiSubject.update({
       where: { id: bpiSubject.id },
-      data: this.mapper.map(bpiSubject, getType<BpiSubject>(), {
-        exclude: ['id'],
-      }) as BpiSubject,
+      data: this.mapper.map(bpiSubject, BpiSubject, BpiSubject),
```

Comment:
do we need a map here as both are domain objects?

Reply:
as we discussed in slack, we are mapping from Prisma model to domain